### PR TITLE
implement criterion Profiler

### PR DESCRIPTION
**Pull request by YangKeao**
Signed-off-by: Yang Keao <keao.yang@yahoo.com>

In this PR, I provide an implementation of `criterion::Profiler` in the `pprof-rs`, so that the users of `pprof-rs` and `criterion` would be easier to integrate and benefit from them.

Inspired by criterion, maybe we can generate [Differential Flame Graphs](http://www.brendangregg.com/blog/2014-11-09/differential-flame-graphs.html) for two benchmarks in the future, which would be really awesome (though `criterion` doesn't support it yet).

### CODE REVIEW

**Comment by YangKeao** [comment]
Hi @jibbow ! Thanks for your article [Automatic Flamegraphs for Benchmarks in Rust](https://www.jibbow.com/posts/criterion-flamegraphs/). I realize that providing an implementation of `criterion::Profiler` inside the `pprof-rs` would be more helpful (and cool :smile_cat: ). Can you help me to check/test this PR or do you have more advice on this?

**Comment by Jibbow** [comment]
Wow that's awesome!! Thanks for pinging me on that :blush: 
Adding this directly to `pprof-rs` is a nice idea, love it! It feels good that my blog post made it into the project :innocent: 

I will add this to my post when this PR has been merged.

File: src/lib.rs [hunk 1]
```diff
@@ -49,3 +51,6 @@ pub mod protos {
 
     include!(concat!(env!("OUT_DIR"), "/perftools.profiles.rs"));
 }
+
+#[cfg(feature = "flamegraph")]
```
Comment (Hexilee) [inline src/lib.rs:55]: Should it be "criterion"?

**Review by Hexilee — COMMENTED** [review]
Rest LGTM

File: src/lib.rs (hunk 1 above)
Comment (YangKeao) [reply src/lib.rs]: Yes :crying_cat_face:

Comment (YangKeao) [reply src/lib.rs]: Fixed.

**Review by Hexilee — APPROVED** [review]
LGTM

**Comment by Hexilee** [comment]
I don't have the writing accessibility to this repo.

**Comment by YangKeao** [comment]
> I don't have the writing accessibility to this repo, maybe you can merge it forcibly.

Now, you have :smiling_imp:

**Comment by Jibbow** [comment]
FYI I updated my blog post and and it now includes a reference to the new criterion profiler in pprof :blush: 
Thanks again!

**Pull request merged**